frontends/p4/validateParsedProgram.h

```diff
@@ -88,7 +88,7 @@ class ValidateParsedProgram final : public Inspector {
     void postorder(const IR::P4Parser *parser) override {
         auto start = parser->states.getDeclaration(IR::ParserState::start);
         if (!start) {
-            ::P4::error(ErrorType::ERR_INVALID, "Parser %1% has no 'start' state", parser);
+            error(ErrorType::ERR_INVALID, "Parser %1% has no 'start' state", parser->name);
```

Comment:
Made a comment but it got lost? 

I would recommend using originalName here to avoid exposing original names. Or just drop the `Parser` in the text.

Reply:
No, the error printer will use `name.originalName` automatically when it prints the error message.  Using `name.originalName` manually in the call will lose the line number information present in the `IR::ID`

Reply:
Thanks